src/api/api.ts

```diff
@@ -0,0 +1,31 @@
+import type { Comments, Product, Products } from "./apiTypes";
+
+const BASE_URL = "https://panda-market-api.vercel.app";
+
+export async function getProducts({ page=1, pageSize=10, orderBy="recent", keyword='' }: {
+    page?: number;
+    pageSize?: number;
+    orderBy?: "recent" | "favorite";
+    keyword?: string;
+}): Promise<Products> {
+    const response = await fetch(`${BASE_URL}/products?page=${page}&pageSize=${pageSize}&orderBy=${orderBy}${keyword && "&keyword="+keyword}`);
+    if (!response.ok) { throw Error("Request Error"); }
+    return (await response.json()) as Products;
+}
```

Comment:
💬 여담
에러 메시지를 더 구체적으로 적으시면 에러가 났을 때 파악하기 더 좋습니다.
```suggestion
export async function getProducts({ page=1, pageSize=10, orderBy="recent", keyword='' }: {
    page?: number;
    pageSize?: number;
    orderBy?: "recent" | "favorite";
    keyword?: string;
}): Promise<Products> {
    const response = await fetch(`${BASE_URL}/products?page=${page}&pageSize=${pageSize}&orderBy=${orderBy}${keyword && "&keyword="+keyword}`);
    
    if (!response.ok) {
      const errorText = await response.text(); // 서버에서 온 응답을 활용
      throw new Error(`Failed to fetch products: ${response.status} - ${errorText}`);
    }
    
    const data: Products = await response.json();
    return data;
}
```